crates/ra_hir/src/ty/primitive.rs

```diff
@@ -2,6 +2,56 @@ use std::fmt;
 
 use crate::{Name, KnownName};
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Copy)]
+pub enum UncertainIntTy {
+    Unknown,
+    Unsigned(UintTy),
+    Signed(IntTy),
+}
```

Comment:
@marcusklaas @flodiebold curious, why this representation was chosen? Wouldn't something like

```
type UncertainIntTy = Option<IntTy>;

struct IntTy {
   signednes:  Signedness,
   bits: IntBits
}

enum Signedness { SIgined, Unsigned }
struct IntBits { X8, X16, X32, X64, X128, Xsize  }
```

be "better" ?

Reply:
Lack of insight, I guess ^^ Your suggestion does look better.

Reply:
:+1: 

filed https://github.com/rust-analyzer/rust-analyzer/issues/1004

cc https://github.com/rust-analyzer/rust-analyzer/pull/1003